ddspipe_participants/src/cpp/reader/rtps/CommonReader.cpp

```diff
@@ -481,6 +481,13 @@ void CommonReader::on_sample_rejected(
                       << ". Reason: " << reason_str);
 }
 
+void CommonReader::on_incompatible_type(
+        fastrtps::rtps::RTPSReader* reader) noexcept
+{
+    logWarning(DDSPIPE_RTPS_COMMONREADER_LISTENER,
```

Comment:
Make the log message a bit more robust. Instead of "shows incompatible type" you could get inspiration from the header's comment: "Reader *this discovered a Writer with a matching Topic  but with an incompatible type".